src/entity/Wallet.java

```diff
@@ -1,177 +1,159 @@
-    private void writeWallet()
-    {
-        try {
-            FileManager.getInstance().saveWallet(this);
-        } catch (IOException e) {
-            logger.warning("Couldn't load wallet!");
-        }
+    private void saveShopToServer() {
+        shopService.saveShop(coin, bulletLevel, shootLevel, livesLevel, coinLevel,
+                _ -> logger.info("Wallet data saved to server successfully."),
+                _ -> logger.warning("Error saving shop data."));
     }
 
-    // 파일에 적힌 정보로 지갑 생성. 만약 파일이 손상되어 읽을 수 없다면 초기값(0)으로 생성하기
-    //Create a wallet using the information written in the file. If the file is damaged or unreadable, create it with initial values (0)
-    public static Wallet getWallet() {
-        BufferedReader bufferedReader = null;
-
-        try {
-            //FileManager 를 통해 파일에서 지갑 데이터를 불러옴
-            //Load wallet data from the file via FileManager
-            bufferedReader = FileManager.getInstance().loadWallet();
-
-            if (bufferedReader == null) {
-                logger.info("Wallet file does not exist, initializing with default values.");
-
-                return new Wallet();
-            }
-
-            //파일에서 각 줄을 읽어와서 값 설정
-            //Read each line from the file and set the values
-            int coin = Integer.parseInt(bufferedReader.readLine());
-            int[] levelSeq = new int[4]; //bulletLevel, shootLevel, livesLevel, coinLevel
-            for (int i = 0; i < 4; i++) {
-                int level = Integer.parseInt(bufferedReader.readLine());
-                if(level > 4 || level <= 0){
-                    logger.info("Weird level. Initializing with default values.");
-                    return new Wallet();
+    private void fetchShopData() {
+        shopService.callShop(
+                (event) -> {
+                    if (event.body() instanceof message.Wallet wallet) {
+                        synchronized (initializationLock) {
+                            this.coin = wallet.coin();
+                            this.bulletLevel = wallet.bulletLevel();
+                            this.shootLevel = wallet.shootLevel();
+                            this.livesLevel = wallet.livesLevel();
+                            this.coinLevel = wallet.coinLevel();
+                            this.initialized = true;
+                            initializationLock.notifyAll();
+                        }
+
+                        logger.info(String.format("Wallet data loaded from server: coin=%d, bulletLevel=%d, shootLevel=%d, livesLevel=%d, coinLevel=%d",
+                                coin, bulletLevel, shootLevel, livesLevel, coinLevel));
+                    } else {
+                        logger.warning("Unexpected response type: " + event.body().getClass().getName());
+                    }
+                },
+                (error) -> {
+                    logger.warning("Error loading shop data: " + error.message());
```

Comment:
Don't use lock.
Follow bellow process
1. when user login, call wallet.initialize
2. Before initialize, When you opened shop screen, Just show default value
3. After initialize, Automatically update screen by `updateEntity` method